lib/services/access_list.go

```diff
@@ -276,3 +280,57 @@ func UnmarshalAccessListReview(data []byte, opts ...MarshalOption) (*accesslist.
 	}
 	return &review, nil
 }
+
+func MatchAccessList(al *accesslist.AccessList, search string) bool {
+	search = strings.ToLower(strings.TrimSpace(search))
+	if search == "" {
+		return true
+	}
+
+	searchTerms := strings.Split(search, " ")
+
+	// Title match
+	if matchesAllTerms(searchTerms, strings.ToLower(al.Spec.Title)) {
+		return true
+	}
+
+	// Name match
+	if matchesAllTerms(searchTerms, strings.ToLower(al.GetName())) {
+		return true
+	}
+
+	// Owner names match
+	for _, owner := range al.Spec.Owners {
+		if matchesAllTerms(searchTerms, strings.ToLower(owner.Name)) {
+			return true
+		}
+	}
+
+	// Description match
+	if matchesAllTerms(searchTerms, strings.ToLower(al.Spec.Description)) {
+		return true
+	}
+
+	// Roles match
+	for _, role := range al.Spec.Grants.Roles {
+		if matchesAllTerms(searchTerms, strings.ToLower(role)) {
+			return true
+		}
+	}
+
+	// Type-specific matches
+	if strings.Contains(al.Origin(), search) {
```

Comment:
Should we apply strings.ToLower here? At the moment, all values are lowercase, but we can’t guarantee that someone won’t introduce uppercase origins in the future.